sklearn/metrics/cluster/_supervised.py

```diff
@@ -47,10 +49,19 @@ def check_clusterings(labels_true, labels_pred):
     labels_true = check_array(
         labels_true, ensure_2d=False, ensure_min_samples=0, dtype=None,
     )
+
     labels_pred = check_array(
         labels_pred, ensure_2d=False, ensure_min_samples=0, dtype=None,
     )
 
+    type_label = type_of_target(labels_true)
+    type_pred = type_of_target(labels_pred)
```

Comment:
A random thought (does not block this PR): We sure call `np.unique` a bunch. (`type_of_target` calls `np.unique` internally).

Reply:
> A random thought (does not block this PR): We sure call np.unique a bunch. (type_of_target calls np.unique internally).

Yes, this is a problem of not using objects... ideally, instead of `unique_labels` and `type_of_target` we should have a `Target` class that reuses partial results such as `unique`, to improve efficiency.